crates/cli/src/cli/simnet/mod.rs

```diff
@@ -39,6 +40,40 @@ struct CheckVersionResponse {
     pub deprecation_notice: Option<String>,
 }
 
+/// Check terminal compatibility and warn about potential display issues
+fn check_and_warn_terminal_compatibility(simnet_events_tx: &Sender<SimnetEvent>, ctx: &Context) {
+    let term_program = env::var("TERM_PROGRAM").unwrap_or_default();
+    let term = env::var("TERM").unwrap_or_default();
+    let colorterm = env::var("COLORTERM").unwrap_or_default();
+
+    // Detect macOS Terminal specifically
+    if term_program == "Apple_Terminal" {
+        let _ = simnet_events_tx.send(SimnetEvent::info(
+            "macOS Terminal detected - using compatibility color mode. For best experience, consider using iTerm2 or similar modern terminal.".to_string()
+        ));
+        info!(
+            ctx.expect_logger(),
+            "Terminal compatibility: macOS Terminal detected, using compatible color palette"
+        );
+    } else if term.starts_with("xterm") && !term.contains("256") && colorterm.is_empty() {
```

Comment:
Having these strings be named constants would help. In this context, I have no idea of the relevance of "256", but a named var would fix that